streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java

```diff
@@ -264,6 +261,79 @@ public void shouldSupportNonPrefixedProducerConfigs() {
         assertEquals(1, configs.get(ProducerConfig.METRICS_NUM_SAMPLES_CONFIG));
     }
 
+    @Test
+    public void shouldForwardCustomConfigsWithNoPrefixToAllClients() {
+        final StreamsConfig streamsConfig = new StreamsConfig(props);
+        props.put("custom.property.host", "host");
+        final Map<String, Object> consumerConfigs = streamsConfig.getConsumerConfigs("groupId", "clientId");
+        final Map<String, Object> restoreConsumerConfigs = streamsConfig.getRestoreConsumerConfigs("clientId");
+        final Map<String, Object> producerConfigs = streamsConfig.getProducerConfigs("clientId");
+        final Map<String, Object> adminConfigs = streamsConfig.getAdminConfigs("clientId");
+        assertEquals("host", consumerConfigs.get("custom.property.host"));
+        assertEquals("host", restoreConsumerConfigs.get("custom.property.host"));
+        assertEquals("host", producerConfigs.get("custom.property.host"));
+        assertEquals("host", adminConfigs.get("custom.property.host"));
+    }
+
+    @Test
+    public void shouldOverrideNonPrefixedCustomConfigsWithPrefixedConfigs() {
+        final StreamsConfig streamsConfig = new StreamsConfig(props);
+        props.put("custom.property.host", "host");
+        props.put(consumerPrefix("custom.property.host"), "localhost");
+        props.put(producerPrefix("custom.property.host"), "localhost");
+        props.put(adminClientPrefix("custom.property.host"), "localhost");
```

Comment:
I think, we should use three different values to make sure that the different prefixes overwrite the configs for the corresponding clients. 

Looking into the test below, they seem to be redundant with this one? We can also remove this test and keep the other three (that would avoid redundancy, too)

Reply:
Yes, using three different values is definitely safer. In addition I think that actually makes the other three tests redundant because it both checks that prefixed properties override non-prefixed ones, and also that prefixed properties are only sent to the specified client. I'm therefore changing this test as suggested and removing the other three. Let me know if you think I should keep them instead.